xbmc/platform/linux/SysfsPath.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright (C) 2011-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/log.h"
+
+#include <fstream>
+#include <string>
+
+class CSysfsPath
+{
+public:
+  CSysfsPath() = default;
+  CSysfsPath(const std::string& path) : m_path(path) {}
+  ~CSysfsPath() = default;
+
+  void operator=(const std::string& path) { m_path = path; }
+
+  template<typename T>
+  bool Get(T& value)
+  {
+    return GetPrivate(value);
```

Comment:
Why the indirection?

Reply:
This is done so I don't have to have duplicated code in the `::Set(std::string)` specialization. I mirrored it for `::Get()` for uniformity. I can get rid of it though for `::Get()` if it's not wanted.